roadmap/implementers-guide/src/runtime/router.md

```diff
@@ -19,6 +17,91 @@ RelayDispatchQueues: map ParaId => Vec<UpwardMessage>;
 RelayDispatchQueueSize: map ParaId => (u32, u32);
 /// The ordered list of `ParaId`s that have a `RelayDispatchQueue` entry.
 NeedsDispatch: Vec<ParaId>;
+/// The downward messages addressed for a certain para.
+DownwardMessageQueues: map ParaId => Vec<DownwardMessage>;
+```
+
+### HRMP
+
+HRMP related structs:
+
+```rust,ignore
+/// A type used to designate a HRMP channel between a (sender, recipient).
+type HrmpChannelId = (ParaId, ParaId);
+
+/// A description of a request to open an HRMP channel.
+struct HrmpOpenChannelRequest {
+	/// The sender and the initiator of this request.
+    sender: ParaId,
+	/// The recipient of the opened request.
+    recipient: ParaId,
+	/// Indicates if this request was confirmed by the recipient.
+    confirmed: bool,
+	/// How many session boundaries this request has seen.
+    age: u32,
+	/// The amount that the sender supplied at the time of creation of this request.
+    sender_deposit: Balance,
+	/// The maximum number of messages that can be pending in the channel at once.
+    limit_used_places: u32,
+	/// The maximum total size of the messages that can be pending in the channel at once.
+    limit_used_bytes: u32,
+}
+
+/// A description of a request to close an opened HRMP channel.
+struct HrmpCloseChannelRequest {
+	/// The para which initiated closing an existing channel.
+    /// invariant: equals either to sender or recipient.
+    initiator: ParaId,
+	/// The identifier of an HRMP channel to be closed.
+    id: HrmpChannelId,
+}
+
+/// A metadata of an HRMP channel.
+struct HrmpChannel {
+    /// The amount that the sender supplied as a deposit when opening this channel.
+    sender_deposit: Balance,
+	/// The amount that the recipient supplied as a deposit when accepting opening this channel.
+    recipient_deposit: Balance,
```

Comment:
What is the actual deposit mechanism? What actions are prevented and how might the accounting work in practice?